keel-core/src/main/kotlin/com/netflix/spinnaker/keel/caffeine/CacheFactory.kt

```diff
@@ -66,16 +88,50 @@ class CacheFactory(
         .recordStats()
     }
 
-  private fun <K, V, C: AsyncCache<K, V>> C.monitor(cacheName: String) =
+  private fun <K, V, C : AsyncCache<K, V>> C.monitor(cacheName: String) =
     CaffeineCacheMetrics.monitor(meterRegistry, this, cacheName)
 }
 
-fun <K, V> (suspend (K) -> V?).toAsyncCacheLoader() : AsyncCacheLoader<K, V> =
+fun <K : Any, V> (suspend (K) -> V?).toAsyncCacheLoader(): AsyncCacheLoader<K, V> =
   AsyncCacheLoader<K, V> { key, executor ->
     CoroutineScope(executor.asCoroutineDispatcher())
       .future { this@toAsyncCacheLoader.invoke(key) }
   }
 
+fun <K : Any, V> (suspend () -> Map<K, V>).toAsyncBulkCacheLoader(): AsyncCacheLoader<K, V> =
+  object : AsyncCacheLoader<K, V> {
+    override fun asyncLoad(key: K, executor: Executor): CompletableFuture<V> {
+      throw UnsupportedOperationException()
+    }
+
+    override fun asyncLoadAll(
+      keys: Iterable<K>,
+      executor: Executor
+    ): CompletableFuture<Map<K, V>> =
+      CoroutineScope(executor.asCoroutineDispatcher())
```

Comment:
The Caffeine library could really use kotlin support so we don't have to do this suspend-function-to-Java future dance.

Reply:
Yeah, I wonder if it would be worth us contributing that